catalog/files/api/src/main/java/org/projectnessie/catalog/files/api/BackendErrorCode.java

```diff
@@ -15,24 +15,13 @@
  */
 package org.projectnessie.catalog.files.api;
 
-import java.io.IOException;
-import java.time.Instant;
-import java.util.Optional;
-
-public abstract class ObjectIOException extends IOException {
-  public ObjectIOException(Throwable cause) {
-    super(cause);
-  }
-
-  public ObjectIOException(String message) {
-    super(message);
-  }
-
-  public ObjectIOException(String message, Throwable cause) {
-    super(message, cause);
-  }
-
-  public abstract boolean isRetryable();
-
-  public abstract Optional<Instant> retryNotBefore();
+public enum BackendErrorCode {
+  UNKNOWN,
+  THROTTLED,
+  NOT_FOUND,
+  UNAUTHORIZED,
+  FORBIDDEN,
+  BAD_REQUEST,
+  NESSIE_ERROR,
+  ICEBERG_ERROR,
```

Comment:
```suggestion
public enum BackendErrorCode {
  /** Unknown errors are retryable, because we do not know the exact reason, only that we do not know the code. */
  UNKNOWN(true),
  THROTTLED(true),
  NOT_FOUND(false),
  UNAUTHORIZED(false),
  FORBIDDEN(false),
  BAD_REQUEST(false),
  NESSIE_ERROR(false),
  ICEBERG_ERROR(false),
  ;

  private final boolean retryable;

  BackendErrorCode(boolean retryable) {
    this.retryable = retryable;
  }

  public boolean retryable() {
    return retryable;
  }
```

Reply:
I moved the `retryable` flag here, but I kept it `false` for `UNKNOWN` - rationale in code comments.